include/seqan3/search/dream_index/interleaved_bloom_filter.hpp

```diff
@@ -0,0 +1,415 @@
+     */
+    inline size_t size() const noexcept
+    {
+        return data.size();
+    }
+    //!\}
+
+    /*!\brief Increases the number of bins stored in the Interleaved Bloom Filter.
+     * \param[in] new_bins The new number of bins.
+     * \throws std::invalid_argument If passed number of bins is smaller than current number of bins.
+     *
+     * \attention This function is only available for **uncompressed** Interleaved Bloom Filters.
+     * \attention The new number of bins must be greater or equal to the current number of bins.
+     *
+     * \details
+     *
+     * The resulting `seqan3::interleaved_bloom_filter` has an increased size proportional to the increase in the
+     * `bin_width` (the number of 64-bit words needed to represent `bins` many bins), e.g.
+     * resizing a `seqan3::interleaved_bloom_filter` with 40 bins to 73 bins also increases the `bin_width` from 1 to
+     * 2 and hence the new `seqan3::interleaved_bloom_filter` will be twice the size.
+     * This increase in size is necessary to avoid invalidating all computed hash functions.
+     * If you want to add more bins while keeping the size constant, you need to rebuild the
+     * `seqan3::interleaved_bloom_filter`.
+     *
+     * ### Example
+     *
+     * \include test/snippet/search/dream_index/interleaved_bloom_filter_increase_bin_number_to.cpp
+     */
+    inline void increase_bin_number_to(size_t const new_bins)
+    //!\cond
+        requires data_layout_mode == data_layout::uncompressed
+    //!\endcond
+    {
+        if (new_bins < bins)
+            throw std::invalid_argument{"The value of new_bins must be >= the current number of bins."};
+
+        // Equivalent to ceil(new_bins / 64)
+        size_t new_bin_width = (new_bins + 63) >> 6;
+
+        if (new_bin_width == bin_width) // No need for internal resize if bin_width does not change.
+        {
+            bins = new_bins;
+            return;
+        }
+
+        size_t new_block_size = new_bin_width << 6;
+        size_t new_bits       = block_count * new_block_size;
+
+        size_t idx_{new_bits}, idx{data.size()};
+        size_t delta = new_block_size - block_size + 64;
+
+        data.resize(new_bits);
+
+        for (size_t i = idx_, j = idx; j > 0; i -= new_block_size, j -= block_size)
+        {
+            size_t stop = i - new_block_size;
+
+            for (size_t ii = i - delta, jj = j - 64; stop && ii >= stop; ii -= 64, jj -= 64)
+            {
+                uint64_t old = data.get_int(jj);
+                data.set_int(jj, 0);
+                data.set_int(ii, old);
+            }
+        }
+
+        bins = new_bins;
+        bin_width = new_bin_width;
+        block_size = new_block_size;
+    }
+
+    /*!\name Comparison operators
+     * \{
+     */
+    /*!\brief Test for equality.
+     * \param[in] lhs A `seqan3::interleaved_bloom_filter`.
+     * \param[in] rhs `seqan3::interleaved_bloom_filter` to compare to.
```

Comment:
If you have an A in the first line, shouldn't it also have an A in the second line? So, A `seqan3::interleaved_bloom_filter` to compare to?